engine/class_modules/monk/sc_monk.cpp

```diff
@@ -7929,6 +8313,109 @@ void monk_t::create_buffs()
+
+  buff.inner_compass_tiger_stance = make_buff_fallback( talent.conduit_of_the_celestials.inner_compass->ok(), this,
+                                                        "tiger_stance", find_spell( 443575 ) );
+
+  buff.jade_sanctuary = make_buff_fallback( talent.conduit_of_the_celestials.jade_sanctuary->ok(), this,
+                                            "jade_sanctuary", find_spell( 448508 ) );
+
+  buff.strength_of_the_black_ox = make_buff_fallback( talent.conduit_of_the_celestials.strength_of_the_black_ox->ok(), this,
+                                                "strength_of_the_black_ox", find_spell( 443112 ) )
+                                      ->set_expire_callback( [ this ]( buff_t *buff_, double, timespan_t ) {
+                                        if ( specialization() == MONK_MISTWEAVER )
+                                        {
+                                          active_actions.strength_of_the_black_ox_absorb->execute();
+                                        }
+                                        else
+                                        {
+                                          active_actions.strength_of_the_black_ox_dmg->execute();
+                                          buff.teachings_of_the_monastery->trigger(
+                                              talent.conduit_of_the_celestials.strength_of_the_black_ox->effectN( 3 )
+                                                  .base_value() );
+                                        }
+                                      } );
+
+  buff.unity_within = make_buff_fallback( talent.conduit_of_the_celestials.unity_within->ok(), this, "unity_within",
+                                          find_spell( 443592 ) )
+                          ->set_expire_callback( [ this ]( buff_t *buff_, double, timespan_t ) {
+                            buff.jade_sanctuary->trigger();
+                            if ( specialization() == MONK_MISTWEAVER )
+                            {
+                              active_actions.flight_of_the_red_crane_celestial_heal->execute();
+                              active_actions.strength_of_the_black_ox_absorb->execute();
+                            }
+                            else
+                            {
+                              active_actions.strength_of_the_black_ox_dmg->execute();
+                              buff.teachings_of_the_monastery->trigger(
+                                talent.conduit_of_the_celestials.strength_of_the_black_ox->effectN( 3 ).base_value() );
+
+                              active_actions.flight_of_the_red_crane_celestial_damage->execute();
+                            }
+                            buff.heart_of_the_jade_serpent_cdr_celestial->trigger();
+                            active_actions.courage_of_the_white_tiger->execute();
+                          } );
```

Comment:
Each of these referenced spell IDs, I would save alongside the talent that provides the spell.

I.e. for `buff.unity_within`, add an object `const spell_data_t *unity_within_buff` immediately following `player_talent_t unity_within`, and initialize it immediately following `unity_within` initialization as well.

Keeps all relevant spells together, and makes it easier to cross-reference all IDs used within an effect to make sure they are the version used in game.